src/scripts/js-telephone-input.js

```diff
@@ -0,0 +1,487 @@
+                switch(reason) {
+                    case 'too-short' :
+                        this.validations.show(this.validations.elements.min, true, false);
+                        break;
+                    case 'too-long' :
+                        this.validations.show(this.validations.elements.max, true, false);
+                        break;
+                    case 'unknown' :
+                        //validations.show(validations.elements.notANumber, true, false);
+                        break;
+                    case 'is-possible' :
+                        this.validations.show(this.validations.elements.possible, true, false);
+                        break;
+                }
+                this.validations.show(this.validations.info.zero, false);
+                //syi.validationsFixer.fixValidationsPosition();
+            }
+        }
+    };
+
+    // Get an example number for the country
+    jsTelephoneInput.prototype.getExampleNumber = function(countryCode) {
+        return this.LibPhoneNumber.getExample(countryCode, 'fixed-line').getNumber('national');
+    };
+
+    //----------------------------------------------------
+    // Private Methods
+    //----------------------------------------------------
+
+    // Format the telephone and execute validations
+    function format(event, component) {
+        if (component.preventKeyup) {
+            component.preventKeyup = false;
+            return;
+        }
+        component.phoneNumber = component.LibPhoneNumber(component.field.value, component.parameters.country);
+        var telephone = component.phoneNumber.getNumber('national');
+        if(telephone) {
+            component.field.value = telephone;
+        }
+        component.validate();
+    }
+
+    // Mix default parameters with the user parameters
+    function extendDefaults(parameters, field) {
+        var defaults = {
+            country: 'ar',
+            required: true,
+            withFlag: true,
+            canChangeCountry: true,
+            messages: {
+                'required': 'Completa este dato.',
+                'valid': 'El teléfono es correcto.',
+                'min': 'Este teléfono tiene menos dígitos de los requeridos.',
+                'invalidDefaultNumber': 'El teléfono cargado en tu cuenta no es válido. Verifícalo por favor.',
+                'max': 'Este teléfono excede el máximo de dígitos posible.',
+                'numbers': 'Sólo puedes ingresar números.',
+                'possible': 'Revisa tu teléfono.',
+                'notANumber': 'El teléfono no es un número.',
+                'zero': 'Ingrésalo siguiendo el formato de este ejemplo: ##example##'
+            },
+            validations : 'telValidations',
+            areaCode : "telAreaCode",
+            number : "telNumber",
+            fullNumber : "telFullNumber"
+        };
+
+        //Checks parameters written as html attributes
+        for (var parameter in defaults) {
```

Comment:
Para iterar en un objeto, [es más seguro utilizar `Object.keys`](http://stackoverflow.com/a/29004402/250301) para no leer propiedades de forma inesperada (`for..in` entra al `prototype`).